nilearn/decomposition/tests/test_canica.py

```diff
@@ -67,5 +90,34 @@ def test_canica_square_img():
     # Smoke test to make sure an error is raised when no data is passed.
     assert_raises(TypeError, canica.fit)
 
+
+def test_component_sign():
+    # BF for issue #570
+
+    # make data (SVD)
+    rng = np.random.RandomState(0)
+    shape = (20, 10, 1)
+    affine = np.eye(4)
+    components = _make_canica_components(shape)
+
+    # make +ve
+    for mp in components:
+        mp[rng.randn(*mp.shape) > .8] *= -5.
+        assert_less_equal(mp.max(), -mp.min())  # goal met ?
+
+    # synthesize data with given components
+    data = _make_data_from_components(components, affine, shape, rng=rng,
+                                      n_subjects=2)
+    mask_img = nibabel.Nifti1Image(np.ones(shape, dtype=np.int8), affine)
+
+    # run CanICA many times (this is known to produce different results)
+    canica = CanICA(n_components=4, random_state=rng, mask=mask_img)
+    for _ in range(3):
+        canica.fit(data)
+        maps = canica.masker_.inverse_transform(canica.components_).get_data()
+        maps = np.rollaxis(maps, 3, 0)
```

Comment:
It would be cooler / more readable if you used iter_img on the img rather than rollaxis on the data.